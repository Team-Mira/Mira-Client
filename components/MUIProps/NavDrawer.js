import { dark } from '../../styles/theme/M3colors'

export const openedMixin = (theme) => ({
  width: 280,
  border: 0,
  backgroundColor: dark.background,
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.enteringScreen,
  }),
  overflowX: 'hidden',
});

export const closedMixin = (theme) => ({
  border: 0,
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
  backgroundColor: dark.background,
  overflowX: 'hidden',
  width: 80,
  [theme.breakpoints.up('sm')]: {
    width: 80,
  },
});

export const closedDrawerButton = (theme) => ({
  width: 55,
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.leavingScreen,
  }),
});

export const openedDrawerButton = (theme) => ({
  width: 262,
  transition: theme.transitions.create('width', {
    easing: theme.transitions.easing.sharp,
    duration: theme.transitions.duration.enteringScreen,
  })
});
