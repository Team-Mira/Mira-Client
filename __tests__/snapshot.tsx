import renderer from 'react-test-renderer'
import NavDrawer from '../components/Drawer'
import { useSession} from "next-auth/react"
import client, {Session} from "next-auth"


jest.mock('next/router', () => ({
  useRouter() {
      return {
          pathname: '',
          push: () => {return true}
      }
  }
}))

jest.mock('next-auth/react')

test('renders NavDrawer before login like snapshot', async () => {

    (useSession as jest.Mock).mockReturnValue([false, false])
    const component = renderer.create(<NavDrawer/>)
    let tree = component.toJSON()
    expect(component).toMatchSnapshot() //first time test is run, it saves a file with current JSX/HTML output and 
    //later tests all future changes to UI against this picture

    renderer.act(() => {
        tree.props.onMouseEnter();
    })

      // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})