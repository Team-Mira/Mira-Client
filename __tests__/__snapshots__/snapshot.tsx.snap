// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavDrawer before login like snapshot 1`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-12jcsqn-MuiDrawer-docked"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13vimk6-MuiToolbar-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-ictp63-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </button>
    </div>
    <ul
      className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1b95rk1-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="DnsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 13H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zM7 19c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM20 3H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1zM7 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Servers
          </p>
        </div>
      </div>
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1b95rk1-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="WysiwygIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm0 16H5V7h14v12zm-2-7H7v-2h10v2zm-4 4H7v-2h6v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Channels
          </p>
        </div>
      </div>
    </ul>
  </div>
</div>
`;

exports[`renders NavDrawer before login like snapshot 2`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-r9748m-MuiDrawer-docked"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13vimk6-MuiToolbar-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-ictp63-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-d1hgq8-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="DnsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 13H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zM7 19c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM20 3H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1zM7 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Servers
          </p>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-d1hgq8-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="WysiwygIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm0 16H5V7h14v12zm-2-7H7v-2h10v2zm-4 4H7v-2h6v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Channels
          </p>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>
</div>
`;

exports[`renders NavDrawer before login like snapshot 3`] = `
<div
  className="MuiDrawer-root MuiDrawer-docked css-12jcsqn-MuiDrawer-docked"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-13vimk6-MuiToolbar-root"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-ictp63-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1b95rk1-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="DnsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 13H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zM7 19c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM20 3H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h16c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1zM7 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Servers
          </p>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        className="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1b95rk1-MuiButtonBase-root-MuiListItemButton-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
            data-testid="WysiwygIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm0 16H5V7h14v12zm-2-7H7v-2h10v2zm-4 4H7v-2h6v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Channels
          </p>
        </div>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>
</div>
`;
